### DISCUSSION TITLE
Add task with priority?

### DISCUSSION THREAD

**Discussion by saifsmailbox98** [todotxt/todo.txt-cli#378] [Q&A]
Is there a way to assign priority while adding the task?

**Comment by inkarkat** [comment]
Oh, there are several very simple add-ons for that. From the [Todo.sh Add on Directory](https://github.com/todotxt/todo.txt-cli/wiki/Todo.sh-Add-on-Directory#add-and-do--add-and-pri-in-one-step):

* [adda](https://github.com/todotxt/todo.txt-cli/blob/addons/.todo.actions.d/adda) (add with priority A)
* [add](https://github.com/doegox/todo.txt-cli/blob/extras/todo.actions.d/add) (both prioritizing (`add pri a`) or doing (`add do`))
  
Or my own:

* [addpri](https://github.com/inkarkat/todo.txt-cli-ex/blob/master/actions/addpri)

Installation basically means downloading the script and putting it in `~/.todo.actions.d/`; see [Creating and Installing Add ons](https://github.com/todotxt/todo.txt-cli/wiki/Creating-and-Installing-Add-ons).

**Comment by bduplanty** [comment]
Enter surrounded with quotes like “(A) This is an A priority task +help @email”

**Reply by inkarkat** [reply]
That works as long you don't use `-t` / `TODOTXT_DATE_ON_ADD=1`. (The priority has to be at the beginning of the line, and and added date would push it to the right.)
